packages/plugin-ext/src/plugin/preference-registry.ts

```diff
@@ -238,7 +238,7 @@ export class PreferenceRegistryExtImpl implements PreferenceRegistryExt {
     private readonly OVERRIDE_PROPERTY_PATTERN = new RegExp(this.OVERRIDE_PROPERTY);
 
     private parseConfigurationData(data: { [key: string]: any }): { [key: string]: any } {
-        return Object.keys(data).reduce((result: any, key: string) => {
+        return Object.keys(data).filter(key => !key.startsWith('__proto__.')).reduce((result: any, key: string) => {
```

Comment:
Wasn't it a good idea to use `Object.create(null)` as the second param to `.reduce` instead of `{}`?

```js
const a = {}
a.__proto__.injected = 1
Object.prototype.injected === 1

const b = Object.create(null)
b.__proto__.injected = 2 // error: __proto__ is undefined
b.__proto__ = { injected: 2 }
Object.prototype.injected === undefined
```

It would avoid the need for filtering. Line 256 would have to be updated as well.

Reply:
Forget about it, the object might be reused in downstream components, no clue where we would need to do the same. Filtering is good.